### ISSUE TITLE
Looking forward to the official implementations :)

### ISSUE THREAD

**Issue by daisukelab** [alexlioralexli/noncontrastive-ssl#2]
Dear authors,

Thank you for sharing your code. And I'm also looking forward to the notebook.
Now, I'm trying to explain my model in my research using cumulative explained variance with my implementations below.

I would be very helpful if you could comment on any issues if possible.
The example below is an output embedding of my SSL model's encoder ViT with audio spectrogram input.
The feature dimension is rather larger than usual. It's a d=5x768=3840 which concatenates 5 tokens per frequency in the same time frame.

```python
def normalized_singular_value(representations):
    # https://github.com/alexlioralexli/noncontrastive-ssl/blob/69e6ad106c588461b3c120153bd4737045c04330/test_collapse.py#L19-L30
    norms = torch.linalg.norm(representations, dim=1)
    normed_reprs = representations / (1e-6 + norms.unsqueeze(1))
    normed_reprs -= normed_reprs.mean(dim=0, keepdims=True)
    stds = torch.svd(normed_reprs).S
    return stds


def cumulative_explained_variance(nsv):
    nsv_sum = nsv.sum()
    raw_cev = torch.tensor([nsv[:i].sum()/nsv_sum for i in range(1, len(nsv) + 1)])
    # AUC
    auc_cev = 1/len(nsv) * raw_cev.sum()
    return auc_cev, raw_cev

### example: embs is the output of my model
nsv = normalized_singular_value(embs.cpu())
auc, cev = cumulative_explained_variance(nsv)
plt.plot(nsv.cpu().numpy())
plt.yscale('log')
plt.show()

print(auc)  # --> tensor(0.8382)
plt.plot(cev.numpy())
```

normalized_singular_value:
![image](https://github.com/alexlioralexli/noncontrastive-ssl/assets/14831220/f16fb7b5-19dc-4bf2-8e9c-3b8cc79a3331)
(So, it looks like there's no apparent partial dimensional collapse with my outputs; ViT with 86M parameters in my test would be more like ResNet-50 than ResNet-18 in your paper.)

cumulative_explained_variance:
![image](https://github.com/alexlioralexli/noncontrastive-ssl/assets/14831220/faecd508-fbdb-4268-8f6c-b4844c63d998)

Thank you

**Comment by alexlioralexli** [comment]
I just uploaded the plotting notebook at `plot_collapse.ipynb`. I don't think you're plotting exactly what I did in the paper, but it's very close. Based on your plots, I think you're right: no apparent partial dimensional collapse.

**Comment by daisukelab** [comment]
Thank you for your comments!
